opengovsg/postmangovsg
# Release v1.10.0

## Features
- Password Protected Email
  - E2E encryption + password hashing
  - Multipart upload
  - Chunked download
  - Password protected page
- Updates to `postman-templating` module
  - Add XSS options
  - Add templating options
- Update diagram to include telegram
- Use full bot token in Telegram webhook URLs
- UI improvements
- Phone number normalisation

## Tests
- Run through a protected campaign flow
- Save a new bot credential in the app
    - Your bot's webhook URL should be set to `telegramOptions.webhookUrl` + the full bot token. Verify this via `GET https://api.telegram.org/bot<token>/getWebhookInfo`. 


## Deploy Notes
- [x] Add protect column to campaigns
```
ALTER TABLE campaigns ADD COLUMN "protect" boolean NOT NULL DEFAULT 'false';
```
- [ ] API Gateway: replace the current `/{botId}` resource with `/{botToken}`
- [ ] Execution role: 
    - Travis: Delete `$PRODUCTION_TELEGRAM_HANDLER_ROLE` and `$STAGING_TELEGRAM_HANDLER_ROLE` since the Lambda can now use the generic execution role
    - Lambda: Set the execution role to the generic one, then delete the Telegram-related roles
- [ ] Update all current bots to use the new webhook URL scheme
